### PR TITLE
HWS-192 Replace read() with read_with_timeout()

### PR DESCRIPTION
**Pull request by YoniEisipsSNN**
This PR should resolve Jira ticket 192. Changes in agilent_common.jl for both oscilloscopes and Impedance analyzers. Some minor spacing fixes.

### CODE REVIEW

**Comment by mofii** [comment]
@YoniEisipsSNN, This one doesn't merge into `main`.
Is there a PR for the other branch that his PR merges to?

File: src/TcpInstruments.jl [hunk 1]
```diff
@@ -61,6 +61,7 @@ const V = u"V"
 const A = u"A"
 const Hz = u"Hz"
 const SMU_NAN = 9.91e37
+const DEFAULT_TIMEOUT = 0.5
```
Comment (mofii) [inline src/TcpInstruments.jl:64]: Let's make this 5 seconds

**Review by mofii — CHANGES_REQUESTED** [review]
Looks great. There were a lot more `read()`s than I was expecting!
Only comment is to increase the standard timeout value

**Comment by mofii** [comment]
@YoniEisipsSNN This PR has merge conflicts (probably from switching the base branch).
See also the comment of increasing the timeout from 0.5s to 5s.

File: src/instrument.jl [hunk 2]
```diff
@@ -319,6 +319,7 @@ function clear_buffer(instr::AbstractInstrument)
     return nothing
 end
 
+
 function read_with_timeout(instr::AbstractInstrument, timeout_sec=2.8)
```
Comment (mofii) [inline src/instrument.jl:323]: Update `timeout_sec=2.8` to 5 seconds

File: src/TcpInstruments.jl [hunk 3]
```diff
@@ -61,6 +61,7 @@ const V = u"V"
 const A = u"A"
 const Hz = u"Hz"
 const SMU_NAN = 9.91e37
+const DEFAULT_TIMEOUT = 5
```
Comment (mofii) [inline src/TcpInstruments.jl:64]: There's no need to define a global for this, the function `read_with_timeout()` already has a standard value for the default time out.

File: src/impedance_analyzer/Agilent4294A.jl [hunk 4]
```diff
@@ -45,7 +45,7 @@ Returns oscillator (ac) voltage
 """
 function get_volt_ac(ia::Instrument{Agilent4294A})
     write(ia, "POWE?")
-    volt_ac = parse(Float64, read(ia)) * u"V"
+    volt_ac = parse(Float64, read_with_timeout(ia, DEFAULT_TIMEOUT)) * u"V"
```
Comment (mofii) [inline src/impedance_analyzer/Agilent4294A.jl:48]: Remove all cases of `DEFAULT_TIMEOUT` and rely on the functions own default value.

**Review by mofii — CHANGES_REQUESTED** [review]
See the comments within the code.

**Comment by mofii** [comment]
Looks great @gita135 !
I just saw we're changing the read() for the impedance analyzer as well. That means we took the timeout down from infinity to 5 seconds. I worry if there are any corner cases for that old instrument where 5 seconds is not enough.
Long story short, I changed my mind on the standard timeout value -> let's increase it to 10 seconds.

**Comment by gita135** [comment]
> Looks great @gita135 ! I just saw we're changing the read() for the impedance analyzer as well. That means we took the timeout down from infinity to 5 seconds. I worry if there are any corner cases for that old instrument where 5 seconds is not enough. Long story short, I changed my mind on the standard timeout value -> let's increase it to 10 seconds.

Done 👍

**Review by mofii — APPROVED** [review]
Great! 👍

**Pull request merged**